compute/firewall/firewall_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+func TestFirewallSnippets(t *testing.T) {
+	ctx := context.Background()
+	firewallsClient, err := compute.NewFirewallsRESTClient(ctx)
+	if err != nil {
+		t.Errorf("NewInstancesRESTClient: %v", err)
```

Comment:
I think at least these first couple errors should actually be t.Fatalf. If they don't succeed the remainder of the test will not either.